walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -159,39 +161,30 @@
   private void findAndAddGapsInOccupiedTimeRanges(long requestedDuration,
       List<TimePoint> orderedTimePoints,
       Collection<TimeRange> availableTimeRanges) {
-    int index = 0;
-    int startCounter = 0;
-    int endCounter = 0;
+    int index = 1;
+    int unendedTimeRange = 1;
 
     while (index < orderedTimePoints.size()) {
-      // Start a continuously occupied range of time.
-      if (startCounter == 0 && orderedTimePoints.get(index).isTimeRangeStart()) {
-        startCounter++;
-        index++;
-      }
-      // Find a continuously occupied range of time, which needs to be skipped. Such continuously
-      // occupied range of time would contain an equal number of start and end {@code TimePoint}.
-      // This occupied range may look like (TR stands for {@code TimeRange}):
+      // Construct a continuously occupied range of time, which needs to be skipped. Such
+      // continuously occupied range of time would contain an equal number of start and end
+      // {@code TimePoint}. This occupied range may look like (TR stands for {@code TimeRange}):
       //    [start of TR1, start of TR2, end of TR1, start of TR3, end of TR3, end of TR2]
-      while (endCounter < startCounter) {
+      // Available time occurs after the continuously occupied ranges of time.
+      if (unendedTimeRange > 0) {
```

Comment:
Can get rid of this conditional, and then you don't need the special case below of setting `unendedTimeRange = 1;`.

Then you just surround the current `else` case in an `if (unendedTimeRange == 0)`

Reply:
Thank you for pointing this out! I changed the code to fit this logic